### ISSUE TITLE
`Layout/EmptyLineAfterGuardClause` false positive inside a `:ruby` filter

### ISSUE THREAD

**Issue by wilfison** [sds/haml-lint#653]
### Description

When a `:ruby` filter ends with a guard clause followed by a blank line, RuboCop still reports `Layout/EmptyLineAfterGuardClause: Add empty line after guard clause`, even though the blank line is clearly present in the HAML source.

The blank line is dropped during the Ruby extraction step, so the Ruby that RuboCop actually inspects has no empty line after the guard clause.

### Steps to reproduce

Create `bug.haml`:

```haml
- collection.each do |item|
  :ruby
    something = item.some(data)
    next if something.nil?

  %p= something
```

Make sure `Layout/EmptyLineAfterGuardClause` is enabled (`.rubocop.yml`):

```yaml
Layout/EmptyLineAfterGuardClause:
  Enabled: true
```

Run:

```bash
haml-lint bug.haml
```

### Actual behavior

```
bug.haml:4 [W] RuboCop: Layout/EmptyLineAfterGuardClause: Add empty line after guard clause.

1 file inspected, 1 lint detected
```

### Expected behavior

No lint — the blank line after the guard clause is there.

```
1 file inspected, 0 lints detected
```

### Root cause

In `lib/haml_lint/ruby_extraction/chunk_extractor.rb`, `visit_filter` extracts the body of a
`:ruby` filter with:

```ruby
lines = node.text.split("\n")
```

`node.text` does contain the trailing blank line (it ends with `\n\n`), but Ruby's
`String#split` **drops trailing empty strings** unless called with a limit of `-1`:

```ruby
"a\nb\n\n".split("\n")      # => ["a", "b"]           <- trailing blank line lost
"a\nb\n\n".split("\n", -1)  # => ["a", "b", "", ""]   <- preserved
```

As a result, the chunk's end marker is assembled right after the guard clause line, so the
Ruby fed to RuboCop looks like:

```ruby
  next if something.nil?
  haml_lint_marker_7        # <- no empty line in between
```

and `Layout/EmptyLineAfterGuardClause` fires. The non-Ruby filter branch is unaffected
because it counts lines via `node.text.count("\n")` and reads them from the original source.

### Environment

- Haml-Lint version: 0.73.0
- Haml version: 7.2.0
- RuboCop version: 1.88.0
- Ruby version: 4.0.2
